photon-ml/src/main/scala/com/linkedin/photon/ml/data/AvroDataReader.scala

```diff
@@ -403,18 +403,34 @@ object AvroDataReader {
    * @param schemaFields the spark sql schema to apply when reading the record
    * @return column values
    */
-  def readColumnValuesFromRecord(record: GenericRecord, schemaFields: Seq[StructField]) = schemaFields
+  protected[data] def readColumnValuesFromRecord(record: GenericRecord, schemaFields: Seq[StructField]) = schemaFields
     .flatMap { field: StructField => field.dataType match {
-      case IntegerType => Some(Utils.getIntAvro(record, field.name))
+      case IntegerType => checkNull(record, field).orElse(Some(Utils.getIntAvro(record, field.name)))
       case StringType => Some(Utils.getStringAvro(record, field.name, field.nullable))
```

Comment:
This is left over from previous code, where we depend on certain behaviors. For example, the code that generates feature keys depends on null being changed to empty string to form name/term keys properly. We need to make a pass and clean this kind of thing up.